src/services/iiif-parser.js

```diff
@@ -8,12 +8,21 @@ import { getReduxManifest } from './get-redux-manifest';
  * @return {boolean} - Does manifest have a canvases array
  **/
 export function canvasesInManifest(manifest) {
-  return (
-    manifesto
-      .create(manifest)
-      .getSequences()[0]
-      .getCanvases().length > 0
-  );
+  const canvases = manifesto
+    .create(manifest)
```

Comment:
It looks like a number of these services are creating new manifesto instances from a passed in manifest (`hasNextSection` is passing the manifest in from redux).  I wonder if the manifesto `create` method is parsing the manifest.  If that's the case then maybe the manifesto instance could be cached somehow to avoid reparsing the manifest each time one of these methods is called.

Reply:
Yes, I agree. I tried using Redux store to store the parsed manifest and then use it in other places. But this made the testing harder. Let me see whether it can be solved :+1:

Reply:
Yes I agree. I tried using the redux store to keep a single parsed manifest along with the raw manifest. But this made testing harder. Let me try whether I can figure it out :+1: